### ISSUE TITLE
iOS - Fails to build

### ISSUE THREAD

**Issue by erikpena** [ezranbayantemur/react-native-rtmp-publisher#24]
### Describe the bug

When attempting to build an iOS project using this React Native package, I'm getting build failures that point to an issue with 'react-native-rtmp-publisher'—

```sh
** BUILD FAILED **

The following build commands failed:
        SwiftEmitModule normal arm64 Emitting\ module\ for\ react_native_rtmp_publisher (in target 'react-native-rtmp-publisher' from project 'Pods')
(1 failure)

info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### To Reproduce

1. Create a new Typescript-based React Native project: `npx react-native init AwesomeTSProject --template react-native-template-typescript`
2. Install `react-native-rtmp-publisher` and be sure to run `pod install` in the `./ios` subdirectory
3. Follow the iOS specific steps outlined in the [README](https://github.com/ezranbayantemur/react-native-rtmp-publisher#ios)
4. In the terminal, attempt to run the command to build and launch the app: `yarn ios`

### Expected behavior

The app to build successfully and the app to launch.

### Version

```json
"dependencies": {
    ...
    "react": "18.1.0",
    "react-native": "0.70.6",
    ...
    "react-native-rtmp-publisher": "^0.4.6",
    ...
  },
```

### Smartphone info.

- Dev machine: Macbook Pro M1
- Device: iPhone 14
- OS: iOS 16.1

### Screenshots

<img width="1028" alt="Screenshot 2023-01-10 at 3 53 59 PM" src="https://user-images.githubusercontent.com/2187347/211686580-37b42759-2956-43fa-baba-44f47dac61b0.png">

**Comment by erikpena** [comment]
Doing a quick google on the error, I found this which seems to be related—

https://stackoverflow.com/questions/73833458/swiftemitmodule-normal-arm64-emitting-module-for-yogakit-in-target-yogakit

The TL;DR is that builds requires a physical device for building/deploying (which makes sense, I was attempting to test on the Simulator).  After setting my iPhone up for Developer Mode and trying again, I'm getting a new set of errors in Xcode—

<img width="438" alt="Screenshot 2023-01-10 at 10 33 10 PM" src="https://user-images.githubusercontent.com/2187347/211735042-72721b4a-4e4a-4d1f-82f5-fd73f0050c16.png">

**Comment by erikpena** [comment]
Also, FWIW, it would be great if deploying to a Simulator didn't break builds.  It makes sense that this library shouldn't work on Simulator, but if a developer is trying to do a quick-check for the UI/UX, they should be able to deploy and check it out without the ability to do a live stream.

For reference, we previously were doing this with nodemediaclient and we would be presented with a black screen when working with the Simulator.

**Comment by ezranbayantemur** [comment]
@erikpena Actually it should work with simulator too, as like you said black screen. 

Can you try to run a build on the example project? I'll be looking at the issue

**Comment by erikpena** [comment]
@ezranbayantemur yes, I can give V that a shot and will let you know how it goes.

**Comment by erikpena** [comment]
Hey @ezranbayantemur,
So I just gave it a go and it's failing for me.  Here is what I did—

1. Clone the `react-native-rtmp-publisher` repo
2. Run `yarn example` in the terminal
3. Changed directory to `./example/ios` <- this step is missing from the readme
4. Run `pod install` <- this step is missing from the readme
5. Changed directory to `../`
6. Run `yarn ios`

I get the following error message in the terminal which indicates that the build failed—

```
CompileAssetCatalog /Users/erik.pena/Library/Developer/Xcode/DerivedData/RtmpExample-gazthtnwqdlimpczjktknfbkuhyn/Build/Products/Debug-iphonesimulator/RtmpExample.app /Users/erik.pena/github/react-native-rtmp-publisher/example/ios/RtmpExample/Images.xcassets (in target 'RtmpExample' from project 'RtmpExample')
    cd /Users/erik.pena/github/react-native-rtmp-publisher/example/ios
    /Applications/Xcode.app/Contents/Developer/usr/bin/actool --output-format human-readable-text --notices --warnings --export-dependency-info /Users/erik.pena/Library/Developer/Xcode/DerivedData/RtmpExample-gazthtnwqdlimpczjktknfbkuhyn/Build/Intermediates.noindex/RtmpExample.build/Debug-iphonesimulator/RtmpExample.build/assetcatalog_dependencies --output-partial-info-plist /Users/erik.pena/Library/Developer/Xcode/DerivedData/RtmpExample-gazthtnwqdlimpczjktknfbkuhyn/Build/Intermediates.noindex/RtmpExample.build/Debug-iphonesimulator/RtmpExample.build/assetcatalog_generated_info.plist --app-icon AppIcon --compress-pngs --enable-on-demand-resources YES --filter-for-thinning-device-configuration iPhone14,7 --filter-for-device-os-version 16.1 --development-region en --target-device iphone --minimum-deployment-target 11.0 --platform iphonesimulator --compile /Users/erik.pena/Library/Developer/Xcode/DerivedData/RtmpExample-gazthtnwqdlimpczjktknfbkuhyn/Build/Products/Debug-iphonesimulator/RtmpExample.app /Users/erik.pena/github/react-native-rtmp-publisher/example/ios/RtmpExample/Images.xcassets


** BUILD FAILED **


The following build commands failed:
        CompileC /Users/erik.pena/Library/Developer/Xcode/DerivedData/RtmpExample-gazthtnwqdlimpczjktknfbkuhyn/Build/Intermediates.noindex/RtmpExample.build/Debug-iphonesimulator/RtmpExample.build/Objects-normal/x86_64/RtmpExample_vers.o /Users/erik.pena/Library/Developer/Xcode/DerivedData/RtmpExample-gazthtnwqdlimpczjktknfbkuhyn/Build/Intermediates.noindex/RtmpExample.build/Debug-iphonesimulator/RtmpExample.build/DerivedSources/RtmpExample_vers.c normal x86_64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'RtmpExample' from project 'RtmpExample')
(1 failure)

info Run CLI with --verbose flag for more details.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

When I attempt to build in Xcode, I get these error—

<img width="1403" alt="Screenshot 2023-01-13 at 4 12 26 PM" src="https://user-images.githubusercontent.com/2187347/212440433-0dd84642-916c-4156-9b87-9450923ca482.png">

Am I missing any steps or did I do something incorrectly here?

**Comment by ezranbayantemur** [comment]
I'll check the example soon as I can and notify you

**Comment by codal-mpawar** [comment]
@ezranbayantemur Any update on this?

**Comment by ezranbayantemur** [comment]
I'm working on in it @codal-mpawar

**Comment by ezranbayantemur** [comment]
@erikpena I've updated the README.md file for about example project build. There was a misleading introduction for example project, sorry about that and thank you for noticing about it.

Basically you can just clone the repo and `yarn` + `cd example && yarn ios`

When I run the project on example I don't see any problem and like I said you can see the black screen as camera view.
![Screenshot 2023-01-21 at 01 31 47](https://user-images.githubusercontent.com/42299721/213825460-6d7701e4-e1a2-496a-afc7-de3562d83759.png)

But right now, as you said there is a problem on iOS build when add the package to fresh project. Currently I'm working on it.

**Comment by ezranbayantemur** [comment]
@erikpena I've released [0.4.7](https://github.com/ezranbayantemur/react-native-rtmp-publisher/releases/tag/v0.4.7) version and this issue should be fixed.

Can you try with the new version?

**Comment by ezranbayantemur** [comment]
@erikpena I'm closing this issue as resolved. You can open it again if it happens again.
